test/upgrade/servingoperator_postupgrade_test.go

```diff
@@ -55,7 +56,17 @@ func TestKnativeServingPostUpgrade(t *testing.T) {
 		// TODO: We only verify the deployment, but we need to add other resources as well, like ServiceAccount, ClusterRoleBinding, etc.
 		resources.SetKodataDir()
 		defer os.Unsetenv(common.KoEnvKey)
-		targetManifest, expectedDeployments := resources.GetExpectedDeployments(t, &v1alpha1.KnativeServing{})
+		targetManifest, err := common.InstalledManifest(&v1alpha1.KnativeServing{})
+		if err != nil {
+			t.Fatalf("Failed to get the manifest for Knative: %v", err)
+		}
+		targetManifest = targetManifest.Filter(ingress.Filters(&v1alpha1.KnativeServing{}))
+
+		expectedDeployments := []string{}
+		for _, resource := range targetManifest.Filter(mf.ByKind("Deployment")).Resources() {
+			expectedDeployments = append(expectedDeployments, resource.GetName())
+		}
+
```

Comment:
I'm not sure what's going on here

Reply:
I need to add the filters to the manifest before checking the deployments. Otherwise it has the kourier deployments too and complains.